crates/hir-def/src/body/scope.rs

```diff
@@ -22,6 +22,7 @@ pub struct ExprScopes {
 #[derive(Debug, PartialEq, Eq)]
 pub struct ScopeEntry {
     name: Name,
+    hygiene: HygieneId,
```

Comment:
(just noting things here now) This wouldn't be needed if `Name` has its syntax context populated either. Though I guess here it would be duplicated, once in `name: Name` and once in the `Binding` referred to by the `binding: BindingId`